fieldsolver/gridGlue.cpp

```diff
@@ -189,256 +189,116 @@ void feedMomentsIntoFsGrid(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>&
-          weightC = weights.at(2) / weightDenominator;
-          weightR1 = weights.at(3) / weightDenominator;
-          weightR2 = weights.at(4) / weightDenominator;
-
-
-          // Get Arrays
-          cell = momentsGrid.get(ii, jj, kk);
-          cellL1 = momentsGrid.get(ii, jj, kk - 1);
-          cellL2 = momentsGrid.get(ii, jj, kk - 2);
-          cellR1 = momentsGrid.get(ii, jj, kk + 1);
-          cellR2 = momentsGrid.get(ii, jj, kk + 2);
-          swap = swapGrid.get(ii, jj, kk);
-
-          //Blur
-          swap->at(fsgrids::moments::RHOM) = weightC * cell->at(fsgrids::moments::RHOM) + weightL1 * cellL1->at(fsgrids::moments::RHOM) + weightL2 * cellL2->at(fsgrids::moments::RHOM) + weightR1 * cellR1->at(fsgrids::moments::RHOM) + weightR2 * cellR2->at(fsgrids::moments::RHOM);
-          swap->at(fsgrids::moments::RHOQ) = weightC * cell->at(fsgrids::moments::RHOQ) + weightL1 * cellL1->at(fsgrids::moments::RHOQ) + weightL2 * cellL2->at(fsgrids::moments::RHOQ) + weightR1 * cellR1->at(fsgrids::moments::RHOQ) + weightR2 * cellR2->at(fsgrids::moments::RHOQ);
-          swap->at(fsgrids::moments::VX) = weightC * cell->at(fsgrids::moments::VX) + weightL1 * cellL1->at(fsgrids::moments::VX) + weightL2 * cellL2->at(fsgrids::moments::VX) + weightR1 * cellR1->at(fsgrids::moments::VX) + weightR2 * cellR2->at(fsgrids::moments::VX);
-          swap->at(fsgrids::moments::VY) = weightC * cell->at(fsgrids::moments::VY) + weightL1 * cellL1->at(fsgrids::moments::VY) + weightL2 * cellL2->at(fsgrids::moments::VY) + weightR1 * cellR1->at(fsgrids::moments::VY) + weightR2 * cellR2->at(fsgrids::moments::VY);
-          swap->at(fsgrids::moments::VZ) = weightC * cell->at(fsgrids::moments::VZ) + weightL1 * cellL1->at(fsgrids::moments::VZ) + weightL2 * cellL2->at(fsgrids::moments::VZ) + weightR1 * cellR1->at(fsgrids::moments::VZ) + weightR2 * cellR2->at(fsgrids::moments::VZ);
-          swap->at(fsgrids::moments::P_11) = weightC * cell->at(fsgrids::moments::P_11) + weightL1 * cellL1->at(fsgrids::moments::P_11) + weightL2 * cellL2->at(fsgrids::moments::P_11) + weightR1 * cellR1->at(fsgrids::moments::P_11) + weightR2 * cellR2->at(fsgrids::moments::P_11);
-          swap->at(fsgrids::moments::P_22) = weightC * cell->at(fsgrids::moments::P_22) + weightL1 * cellL1->at(fsgrids::moments::P_22) + weightL2 * cellL2->at(fsgrids::moments::P_22) + weightR1 * cellR1->at(fsgrids::moments::P_22) + weightR2 * cellR2->at(fsgrids::moments::P_22);
-          swap->at(fsgrids::moments::P_33) = weightC * cell->at(fsgrids::moments::P_33) + weightL1 * cellL1->at(fsgrids::moments::P_33) + weightL2 * cellL2->at(fsgrids::moments::P_33) + weightR1 * cellR1->at(fsgrids::moments::P_33) + weightR2 * cellR2->at(fsgrids::moments::P_33);
-
-          // Reset Weights
-          weights.fill(0);
         }
       }
     }
-    momentsGrid.updateGhostCells();
+    phiprof::stop("BlurPass");
+
+    phiprof::start("GhostUpdate");
     swapGrid.updateGhostCells();
```

Comment:
Don't we have to update the swapGrid ghost cells at least before copying back to momentsGrid?

Reply:
Wouldn't think so since we do the ghost cell upgrade after replacing the data (l296). But Others can verify.

Reply:
True, we always only loop on the local domain (excluding ghost cells) so it should be safe. It would be different if we were accessing neighbours which could then potentially be ghosts.